### DISCUSSION TITLE
De-interleaving images/signals

### DISCUSSION THREAD

**Discussion by bruno-f-cruz** [bonsai-rx/bonsai#889] [Q&A] [answered]
The following question was asked in the Bonsai Users - Google Groups
[Link to original post](https://groups.google.com/g/bonsai-users/c/4fdMA_n6Gb4)

---
**Original post**

Hi Bonsai community,

Does anyone know a straightforward way to de-interleave images/signals?

I'm using SpinnakerCapture to acquire images from a FLIR BFS at 150Hz and want to online separate it into e.g. 5x 30Hz videos (Frame#: 5n, 5n+1, 5n+2, 5n+3, and 5n+4).

Thanks!
Kenta

**Comment by bruno-f-cruz** [comment, marked as answer]
Hi Kenta,

Would something like this work?

![image](https://user-images.githubusercontent.com/7049351/165741903-021c4d31-8916-4faa-b4b4-76fa54d1079e.png)

`Skip(N)` sets the phase of the stream by skipping the first N samples coming from the camera.
`Buffer(1,S)` grabs a sample every S-1 samples that come from the camera (downsampling step). 
`Index(0)` grabs the first and only element in the buffer

Cheers,
B

**Reply by hagikent** [reply]
Thanks Bruno!
